src/Symfony/Component/Cache/Traits/RedisTrait.php

```diff
@@ -152,10 +152,10 @@ public static function createConnection(string $dsn, array $options = [])
         if (isset($params['host']) || isset($params['path'])) {
             if (!isset($params['dbindex']) && isset($params['path'])) {
                 if (preg_match('#/(\d+)?$#', $params['path'], $m)) {
-                    $params['dbindex'] = $m[1] ?? '0';
+                    $params['dbindex'] = $m[1] ?? (isset($query['dbindex']) ? $query['dbindex'] : '0');
                     $params['path'] = substr($params['path'], 0, -\strlen($m[0]));
                 } elseif (isset($params['host'])) {
-                    throw new InvalidArgumentException('Invalid Redis DSN: query parameter "dbindex" must be a number.');
+                    throw new InvalidArgumentException('Invalid Redis DSN: path parameter "dbindex" must be a number.');
```

Comment:
it can be either so maybe this instead?
```suggestion
                    throw new InvalidArgumentException('Invalid Redis DSN: parameter "dbindex" must be a number.');
```